x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/utils/custom_sort_script.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const VULNERABILITY_SEVERITY_FIELD = 'vulnerability.severity';
+import { i18n } from '@kbn/i18n';
+
+export const severitySchemaConfig = {
+  type: 'severitySchema',
+  detector() {
+    return 0; // this schema is always explicitly defined
+  },
+  sortTextAsc: i18n.translate('xpack.csp.vulnerabilityTable.column.sortAscending', {
+    defaultMessage: 'Low -> Critical',
+  }),
+  sortTextDesc: i18n.translate('xpack.csp.vulnerabilityTable.column.sortDescending', {
+    defaultMessage: 'Critical -> Low',
+  }),
+  icon: 'dot',
+  color: '',
+};
+
+export const severitySortScript = (direction: string) => ({
+  _script: {
+    type: 'number',
+    script: {
+      lang: 'painless',
+      inline:
+        "if(params.scores.containsKey(doc['vulnerability.severity'].value)) { return params.scores[doc['vulnerability.severity'].value];} return 1000;",
```

Comment:
do we have vulnerabilities without `vulnerability.severity`? if that is a possible case, we need to check if that value exists before (otherwise we will get `all shards failed` error)

so incase it doesn't exist lets return `-1`

Reply:
I don't see vulnerabilities without `vulnerability.severity` but I can add an additional check just in case of `vulnerability.severity` is missing.

Reply:
> https://github.com/elastic/kibana/blob/68ca864e96ff213fad16711984f4b4c30741f808/x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/hooks/use_latest_vulnerabilities.tsx#L44
> 
> Unknown `vulnerability.severity` is filtered out.

This is just filtering out UNKNOWN, but now I think we can add it to the `exists` filter, to make sure we won't have vulnerabilities with an empty value.

@Omolola-Akinleye can you add the following to the `getFindingsQuery` function? https://github.com/elastic/kibana/blob/68ca864e96ff213fad16711984f4b4c30741f808/x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/hooks/use_latest_vulnerabilities.tsx#L29

```
{ exists: { field: 'vulnerability.severity' } },
```

Reply:
I think it would be better not to couple between the logic of the filtering and taking it as an assumption in other areas in our code

Checking it exists in painless would make it robust, so in case someone would be asked to change the filter, they won't encounter edge cases in other parts in our code